### PR TITLE
storage: add range keys and mvccstats to gc bench

### PR DESCRIPTION
**Pull request by aliher1911**
Previously gc benchmarking code didn't include neither range key
tombstones, nor mvcc stats calculations.
This is not realistic as production always uses stats and it is
a non-negligible part of the operation time.
This commit adds option to enable stats calculation during
benchmarking as well as adding underlying range key tombstone
to include everything.

Release justification: increasing bench coverage, non production.
Release note: None

### CODE REVIEW

File: pkg/storage/bench_test.go [hunk 1]
```diff
@@ -1526,31 +1539,41 @@ func runMVCCGarbageCollect(
...
 	//
 	// NB: a real invocation of MVCCGarbageCollect typically has most of the keys
 	// in sorted order. Here they will be ordered randomly.
 	setup := func() (gcKeys []roachpb.GCRequest_GCKey) {
 		batch := eng.NewBatch()
+		if opts.numRangeKeys > 1 {
+			b.Fatal("Invalid bench data config. Number of range keys can be 0 or 1")
+		}
+		if opts.numRangeKeys == 1 {
+			if err := MVCCDeleteRangeUsingTombstone(ctx, batch, nil, keys.MinKey.Next(), keys.MaxKey,
```
Comment (erikgrinaker) [inline pkg/storage/bench_test.go:1562]: We don't really allow MVCC range tombstones across local keyspace, but the check is currently a bit loose. Added #87403 to tighten it up. This call should use `Keys.LocalMax` as the start key.

File: pkg/storage/bench_test.go [hunk 2]
```diff
@@ -63,6 +63,8 @@ func BenchmarkMVCCGarbageCollect(b *testing.B) {
 		{2, []int{1}},
 		{1024, []int{1, 16, 32, 512, 1015, 1023}},
 	}
+	numRangeTombstones := []int{0, 1}
```
Comment (erikgrinaker) [inline pkg/storage/bench_test.go:66]: Would be nice to have a variant with 100 random range tombstones too, like the other benchmarks, but this is fine for now.

**Review by erikgrinaker — APPROVED** [review]

File: pkg/storage/bench_test.go (hunk 2 above)
Comment (aliher1911) [reply pkg/storage/bench_test.go:66]: That requires reverting key by key to ts by ts insertion. I did this change when experimenting with batch performance, but I'd rather add it separately.

File: pkg/storage/bench_test.go [hunk 3]
```diff
@@ -83,16 +85,26 @@ func BenchmarkMVCCGarbageCollect(b *testing.B) {
...
-																numKeys:     numKeys,
-																numVersions: versions.total,
-																valueBytes:  valSize,
-															},
-															keyBytes:       keySize,
-															deleteVersions: toDelete,
+													for _, rangeTombstones := range numRangeTombstones {
+														b.Run(fmt.Sprintf("numRangeTs=%d", rangeTombstones), func(b *testing.B) {
+															for _, stats := range updateStats {
+																b.Run(fmt.Sprintf("updateStats=%t", stats), func(b *testing.B) {
```
[the remark this answers is not in the document]
Comment (aliher1911) [reply pkg/storage/bench_test.go:91]: It is helpful when comparing to previous releases mostly. Maybe we should remove unhelpful benchmarks later.
Maybe we should backport benchmarks like that to make life easier?

File: pkg/storage/bench_test.go (hunk 2 above)
Comment (aliher1911) [reply pkg/storage/bench_test.go:66]: Created #87417

File: pkg/storage/bench_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply pkg/storage/bench_test.go:91]: No strong opinion either way. It's fine to keep it in as well.

**Comment by aliher1911** [comment]
TFTR

bors r+

**Pull request merged**